corehq/apps/es/management/commands/resave_failed_forms_and_cases.py

```diff
@@ -14,52 +19,83 @@
 from corehq.form_processor.utils.xform import resave_form
 from corehq.apps.hqcase.utils import resave_case
 from corehq.util.log import with_progress_bar
+from dimagi.utils.chunked import chunked
+
+DATE_FORMAT = "%Y-%m-%d"
+
+
+def valid_date(s):
+    try:
+        return datetime.strptime(s, DATE_FORMAT)
+    except ValueError:
+        msg = "Not a valid date: '{0}'.".format(s)
+        raise argparse.ArgumentTypeError(msg)
 
 
 class Command(BaseCommand):
     def add_arguments(self, parser):
         parser.add_argument('domain')
+        parser.add_argument(
+            'start_date',
+            type=valid_date,
+            help='The start date (inclusive). format YYYY-MM-DD'
+        )
+        parser.add_argument(
+            'end_date',
+            type=valid_date,
+            help='The end date (exclusive). format YYYY-MM-DD'
+        )
         parser.add_argument('--xforms', action='store_true')
         parser.add_argument('--cases', action='store_true')
 
-    def handle(self, domain, *args, **options):
+    def handle(self, domain, start_date, end_date, *args, **options):
         resave_xforms = options.get('xforms')
         resave_cases = options.get('cases')
         if resave_xforms:
-            perform_resave_on_xforms(domain)
+            perform_resave_on_xforms(domain, start_date, end_date)
         if resave_cases:
-            perform_resave_on_cases(domain)
+            perform_resave_on_cases(domain, start_date, end_date)
 
 
-def perform_resave_on_xforms(domain):
-    xform_ids_missing_in_es, _ = compare_xforms(domain, 'XFormInstance')
+def perform_resave_on_xforms(domain, start_date, end_date):
+    _, _, xform_ids_missing_in_es, _ = compare_xforms(domain, 'XFormInstance', start_date, end_date)
     print("%s Ids found for xforms missing in ES." % len(xform_ids_missing_in_es))
-    print(xform_ids_missing_in_es)
+    if len(xform_ids_missing_in_es) < 1000:
+        print(xform_ids_missing_in_es)
     ok = input("Type 'ok' to continue: ")
     if ok != "ok":
         print("No changes made")
         return
     form_accessor = FormAccessors(domain)
-    for xform_id in with_progress_bar(xform_ids_missing_in_es):
-        xform = form_accessor.get_form(xform_id)
-        if xform:
+    for xform_ids in chunked(with_progress_bar(xform_ids_missing_in_es), 100):
+        xforms = form_accessor.get_forms(xform_ids)
+        found_xform_ids = set()
+
+        for xform in xforms:
             resave_form(domain, xform)
-        else:
+            found_xform_ids.add(xform.form_id)
+
+        for xform_id in xform_ids - found_xform_ids:
             print("form not found %s" % xform_id)
 
 
-def perform_resave_on_cases(domain):
-    case_ids_missing_in_es, _ = compare_cases(domain, 'CommCareCase')
+def perform_resave_on_cases(domain, start_date, end_date):
+    _, _, case_ids_missing_in_es, _ = compare_cases(domain, 'CommCareCase', start_date, end_date)
```

Comment:
where do these extra args come from (the `_, _,` at the start)

Reply:
In the same PR linked earlier the return value of this function was changed https://github.com/dimagi/commcare-hq/pull/20999/files#diff-972d6eb693aff7ef3de866a488625cdeR143